src/Forcings/advective_forcing.jl

```diff
@@ -0,0 +1,58 @@
+using Oceananigans.Advection: UpwindBiasedFifthOrder
+using Oceananigans.Fields: ZeroField
+
+struct AdvectiveForcing{U, S, F, C}
+    velocities :: U
+    advection_scheme :: S
+    advection_kernel_function :: F
+    advected_field :: C
+end
+
+"""
+    AdvectiveForcing(scheme=UpwindBiasedFifthOrder(), u=ZeroField(), v=ZeroField(), w=ZeroField())
+
+Build a forcing term representing advection by the velocity field `u, v, w` with an advection `scheme`.
+
+Example
+=======
+
+# Using a tracer field to model sinking particles
+
+```jldoctest
+using Oceananigans
+
+# Physical parameters
+gravitational_acceleration          = 9.81     # m s⁻²
+ocean_density                       = 1026     # kg m⁻³
+mean_particle_density               = 2000     # kg m⁻³
+mean_particle_radius                = 1e-3     # m
+ocean_molecular_kinematic_viscosity = 1.05e-6  # m² s⁻¹
+
+# Terminal velocity of a sphere in viscous flow
+Δb = gravitational_acceleration * (mean_particle_density - ocean_density) / ocean_density
+ν = ocean_molecular_kinematic_viscosity
+R = mean_particle_radius
+
+w_Stokes = - 2/9 * Δb / ν * R^2
+
+settling = AdvectiveForcing(WENO5(), w=w_Stokes)
+"""
+function AdvectiveForcing(scheme=UpwindBiasedFifthOrder(); u=ZeroField(), v=ZeroField(), w=ZeroField())
+    velocities = (; u, v, w)
+    return AdvectiveForcing(velocities, scheme, nothing, nothing) # stub
+end
+    
+function regularize_forcing(af::AdvectiveForcing, field, field_name, model_field_names)
+    kernel_function =
+        field_name === :u ? div_𝐯u :
+        field_name === :v ? div_𝐯v :
+        field_name === :w ? div_𝐯w : div_Uc
```

Comment:
Trying to understand the syntax.

if `field_name` is `u`, then `kernel_function` is `div_vu`,
but if `field_name` is `v`, then `kernel_function` is `div_vv`,
and if `field_name` is `w`, then `kernel_function` is `div_vw`
if is none of that, use `div_Uc`?

Isn't is better to use `ifelse` instead? Does this runs on GPU?

Reply:
No, this function only runs once when the model is constructed, so here we prioritize readability.